### PR TITLE
commit endpoint related ballerina changes

### PR DESCRIPTION
**Pull request by tharindu1st**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.wso2.carbon.apimgt.rest.api.core.factories;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: licence header is missing

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/EndpointsApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            } else {
+                String msg = "Endpoint is not found with apiId : " + endpointId;
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(msg, 900314L, msg);
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).header(HttpHeaders.CONTENT_TYPE,
+                        MediaType.APPLICATION_JSON).entity(errorDTO).build();
+            }
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while retrieving gateway config of Endpoint : " + endpointId;
+            HashMap<String, String> paramList = new HashMap<String, String>();
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/EndpointsApiServiceImpl.java]: cant we use ? Map<String, String> paramList = new HashMap<String, String>();

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/EndpointsApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+public class EndpointsApiServiceImpl extends EndpointsApiService {
+    private static final Logger log = LoggerFactory.getLogger(EndpointsApiServiceImpl.class);
+
+    @Override
+    public Response endpointsEndpointIdGatewayConfigGet(String endpointId
+            , String accept
+            , Request request) throws NotFoundException {
+        try {
+            APIMgtAdminService apiMgtAdminService = APIManagerFactory.getInstance().getAPIMgtAdminService();
+            String endpointGatewayConfig;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/EndpointsApiServiceImpl.java]: can do in a single line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.wso2.carbon.apimgt.rest.api.core.EndpointsApiService;
+import org.wso2.carbon.apimgt.rest.api.core.impl.EndpointsApiServiceImpl;
+
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: class level comment?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+public class EndpointsApiServiceFactory {
+    private static final EndpointsApiService service = new EndpointsApiServiceImpl();
+
+    public static EndpointsApiService getEndpointsApi() {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: what this method do?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/dto/EndpointListDTO.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
+package org.wso2.carbon.apimgt.rest.api.core.dto;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/dto/EndpointListDTO.java]: licence?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/EndpointsApiService.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package org.wso2.carbon.apimgt.rest.api.core;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/EndpointsApiService.java]: licence?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java [hunk 8]
```diff
@@ -1202,6 +1202,50 @@ public void updateRating(String apiId, String ratingId, Rating rating) throws AP
...
     @Override
+    public List<String> getUUIDsOfGlobalEndpoints() throws APIMgtDAOException {
+        final String query = "SELECT UUID FROM AM_ENDPOINT WHERE APPLICABLE_LEVEL='" + APIMgtConstants
+                .GLOBAL_ENDPOINT + "'";
+        List<String> endpointList = new ArrayList<>();
+        try (Connection connection = DAOUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query);
+             ResultSet resultSet = statement.executeQuery()) {
+            while (resultSet.next()) {
+                endpointList.add(resultSet.getString("UUID"));
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java]: move uuid string to a constant

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java [hunk 9]
```diff
@@ -1202,6 +1202,50 @@ public void updateRating(String apiId, String ratingId, Rating rating) throws AP
...
+
+    @Override
+    public String getEndpointConfig(String endpointId) throws APIMgtDAOException {
+        final String query = "SELECT GATEWAY_CONFIG FROM AM_ENDPOINT WHERE UUID = ?";
+        try (Connection connection = DAOUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, endpointId);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    InputStream inputStream = resultSet.getBinaryStream("GATEWAY_CONFIG");
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java]: gateway config to a constant?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java (hunk 1 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: this is a auto generated class therefore it not need to have lisence header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java (hunk 4 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: same for here

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java (hunk 5 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/factories/EndpointsApiServiceFactory.java]: same for here

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/dto/EndpointListDTO.java (hunk 6 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/dto/EndpointListDTO.java]: same here

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/EndpointsApiService.java (hunk 7 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/gen/java/org/wso2/carbon/apimgt/rest/api/core/EndpointsApiService.java]: same here

**Pull request merged**